consensus/ibft/src/test/java/tech/pegasys/pantheon/consensus/ibft/statemachine/RoundChangeManagerTest.java

```diff
@@ -0,0 +1,180 @@
+
+    Map<ConsensusRoundIdentifier, MessageValidator> messageValidators = Maps.newHashMap();
+
+    messageValidators.put(
+        ri1,
+        new MessageValidator(
+            validators,
+            Util.publicKeyToAddress(proposerKey.getPublicKey()),
+            ri1,
+            headerValidator,
+            protocolContext,
+            parentHeader));
+
+    messageValidators.put(
+        ri2,
+        new MessageValidator(
+            validators,
+            Util.publicKeyToAddress(validator1Key.getPublicKey()),
+            ri2,
+            headerValidator,
+            protocolContext,
+            parentHeader));
+
+    messageValidators.put(
+        ri3,
+        new MessageValidator(
+            validators,
+            Util.publicKeyToAddress(validator2Key.getPublicKey()),
+            ri3,
+            headerValidator,
+            protocolContext,
+            parentHeader));
+
+    manager = new RoundChangeManager(2, validators, messageValidators::get);
+  }
+
+  private SignedData<RoundChangePayload> makeRoundChangeMessage(
+      final KeyPair key, final ConsensusRoundIdentifier round) {
+    MessageFactory messageFactory = new MessageFactory(key);
+    return messageFactory.createSignedRoundChangePayload(round, Optional.empty());
+  }
+
+  @Test
+  public void rejectsInvalidRoundChangeMessage() {
+    SignedData<RoundChangePayload> roundChangeData = makeRoundChangeMessage(nonValidatorKey, ri1);
+    assertThat(manager.appendRoundChangeMessage(roundChangeData)).isEqualTo(Optional.empty());
+    assertThat(manager.roundChangeCache.get(ri1)).isNull();
+  }
+
+  @Test
+  public void acceptsValidRoundChangeMessage() {
+    SignedData<RoundChangePayload> roundChangeData = makeRoundChangeMessage(proposerKey, ri2);
+    assertThat(manager.appendRoundChangeMessage(roundChangeData)).isEqualTo(Optional.empty());
+    assertThat(manager.roundChangeCache.get(ri2).receivedMessages.size()).isEqualTo(1);
+  }
+
+  @Test
+  public void doesntAcceptDuplicateValidRoundChangeMessage() {
+    SignedData<RoundChangePayload> roundChangeData = makeRoundChangeMessage(proposerKey, ri2);
+    assertThat(manager.appendRoundChangeMessage(roundChangeData)).isEqualTo(Optional.empty());
+    assertThat(manager.appendRoundChangeMessage(roundChangeData)).isEqualTo(Optional.empty());
+    assertThat(manager.roundChangeCache.get(ri2).receivedMessages.size()).isEqualTo(1);
+  }
+
+  @Test
+  public void becomesReadyAtThreshold() {
+    SignedData<RoundChangePayload> roundChangeDataProposer =
+        makeRoundChangeMessage(proposerKey, ri2);
+    SignedData<RoundChangePayload> roundChangeDataValidator1 =
+        makeRoundChangeMessage(validator1Key, ri2);
+    assertThat(manager.appendRoundChangeMessage(roundChangeDataProposer))
+        .isEqualTo(Optional.empty());
+    assertThat(manager.appendRoundChangeMessage(roundChangeDataValidator1))
+        .satisfies(Optional::isPresent);
+  }
+
+  @Test
+  public void doesntReadyWithDifferentRounds() {
```

Comment:
Is this named appropriately, or is it missing a word?

Reply:
clarified